tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumWebUI.cs

```diff
@@ -82,75 +96,66 @@ private IWebDriver InitDriverAndGoToUrl(string url)
-                        switch (authResult.Status)
-                        {
-                            case AuthorizationStatus.Success:
-                                return CloseWindowSuccessHtml;
-                            default:
-                                return CloseWindowFailureHtml;
-                        }
+                        authCodeUri = uri;
+
+                        return GetMessageToShowInBroswerAfterAuth(uri);
                     },
-                    cancellationTokenSource.Token);
+                    cancellationToken);
 
                 try
                 {
 
                     // Run the tcp listener and the selenium automation in parallel
-                    Task seleniumAutomationTask = Task.Run(() =>
+                    var seleniumAutomationTask = Task.Run(() =>
                     {
                         _seleniumAutomationLogic(driver);
                     });
 
                     await Task.WhenAll(seleniumAutomationTask, listenForAuthCodeTask).ConfigureAwait(false);
-                    return authResult;
-
-                }
-                catch (Exception ex) when (ex is TaskCanceledException || ex is OperationCanceledException)
-                {
-                    var result = new AuthorizationResult(AuthorizationStatus.UserCancel)
-                    {
-                        ErrorDescription = "Listening for an auth code was cancelled or has timed out.",
-                        Error = "system_browser_cancel_or_timeout_exception"
-                    };
+                    return authCodeUri;
 
-                    return await Task.FromResult(result).ConfigureAwait(false);
                 }
+               
                 catch (SocketException ex)
                 {
-                    var result = new AuthorizationResult(AuthorizationStatus.UnknownError)
-                    {
-                        ErrorDescription = ex.Message + " socket error code: " + ex.SocketErrorCode,
-                        Error = "system_browser_socket_exception"
-                    };
-
-                    return await Task.FromResult(result).ConfigureAwait(false);
+                    throw new MsalCustomWebUiFailedException(
+                        "A socket exception occured " + ex.Message + " socket error code " + ex.SocketErrorCode);
                 }
                 catch (Exception ex)
                 {
-                    var result = new AuthorizationResult(AuthorizationStatus.UnknownError)
-                    {
-                        ErrorDescription = ex.Message,
-                        Error = "system_browser_waiting_exception"
-                    };
-
-                    return await Task.FromResult(result).ConfigureAwait(false);
+                    throw new MsalCustomWebUiFailedException("Unkown error: " + ex.Message);
                 }
             }
         }
+
+        private static string GetMessageToShowInBroswerAfterAuth(Uri uri)
```

Comment:
Is it something generic enough that we should provide part of the extensibility?
or just provide guidance?

Reply:
Just provide guidance I think. We should create a wiki page showing how to E2E test apps that use auth.

Reply:
be my guest :)